File: 品味Java的21種設計模式/dp_java/command/example5/DuckCommand.java
package cn.javass.dp.command.example5;
/**
 * 命令对象，北京烤鸭
 */
public class DuckCommand implements Command{
	private CookApi cookApi = null;
	public void setCookApi(CookApi cookApi) {
		this.cookApi = cookApi;
	}
	
	public void execute() {
		this.cookApi.cook("北京烤鸭");
	}
}

File: 品味Java的21種設計模式/dp_cpp/command/example5/DuckCommand.cpp
#include "DuckCommand.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example5
				{

					void DuckCommand::setCookApi(CookApi *cookApi)
					{
						this->cookApi = cookApi;
					}

					void DuckCommand::execute()
					{
						this->cookApi->cook("北京烤鸭");
					}
				}
			}
		}
	}
}
